mmdet/models/dense_heads/detr_head.py

```diff
@@ -87,21 +73,9 @@ def __init__(self,
         # since it brings inconvenience when the initialization of
         # `AnchorFreeHead` is called.
         super(AnchorFreeHead, self).__init__()
-        use_sigmoid_cls = loss_cls.get('use_sigmoid', False)
-        assert not use_sigmoid_cls, 'setting use_sigmoid_cls as True is ' \
-            'not supported in DETR, since background is needed for the ' \
-            'matching process.'
-        assert 'embed_dims' in transformer \
-            and 'num_feats' in positional_encoding
-        num_feats = positional_encoding['num_feats']
-        embed_dims = transformer['embed_dims']
-        assert num_feats * 2 == embed_dims, 'embed_dims should' \
-            f' be exactly 2 times of num_feats. Found {embed_dims}' \
-            f' and {num_feats}.'
-        assert test_cfg is not None and 'max_per_img' in test_cfg
-
+        self.bg_cls_weight = 0
         class_weight = loss_cls.get('class_weight', None)
-        if class_weight is not None:
+        if class_weight is not None and (self.__class__ is DETRHead):
```

Comment:
can we use `isinstance(self, DETRHead)`?

Reply:
We can't set class_weight in a subclass of `DETRHead`, for example, `DeformDETRhead`